aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -619,4 +635,17 @@ public void setConf(Configuration conf) {
   protected Map<String, String> properties() {
     return catalogProperties == null ? ImmutableMap.of() : catalogProperties;
   }
+
+  private Cache<TableOperations, FileIO> newFileIOCloser() {
+    return Caffeine.newBuilder()
+        .weakKeys()
+        .removalListener(
+            (RemovalListener<TableOperations, FileIO>)
+                (ops, fileIO, cause) -> {
+                  if (null != fileIO) {
+                    fileIO.close();
```

Comment:
This is not a regular cache, this could be better served with `java.lang.ref.Cleaner`.
However, none of these addresses the main problem that we use Java GC as a proxy to discover and close resource leaks. I understand that fileIO instances can hold on to resources that require closing (like file descriptors or sockets inside AWS S3 client). If i have abundance of memory, the weakly held instances may not get collected, so the app can run out of descriptors. Can you please help me understand why this isn't a concern?